crates/amalthea/src/socket/shell.rs

```diff
@@ -88,14 +81,14 @@ impl Shell {
         socket: Socket,
         iopub_tx: Sender<IOPubMessage>,
         comm_manager_tx: Sender<CommManagerEvent>,
-        shell_handler: Arc<Mutex<dyn ShellHandler>>,
+        shell_handler: Box<dyn ShellHandler>,
         lsp_handler: Option<Arc<Mutex<dyn ServerHandler>>>,
         dap_handler: Option<Arc<Mutex<dyn ServerHandler>>>,
     ) -> Self {
         Self {
             socket,
             iopub_tx,
-            shell_handler,
+            shell_handler: RefCell::new(shell_handler),
```

Comment:
Maybe create it on its own line, with a comment about the need for RefCell? (because some handler methods take a `mut` self?)